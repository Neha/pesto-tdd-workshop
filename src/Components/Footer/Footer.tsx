import React from 'react';
import { FOOTER } from '../../Constants/Constants';

interface props{}

const Footer : React.FC<props> = () => {
    return(
        <footer className="footerWrapper" >
            <a href="aboutus.html">{FOOTER.ABOUT_US}</a>
            <a href="contact.html">{FOOTER.CONTACT_US}</a>
        </footer>
    )
}

export default Footer;
