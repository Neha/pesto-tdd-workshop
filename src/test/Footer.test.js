import {render, screen } from '@testing-library/react';
import Footer from '../Components/Footer/Footer';

describe("Test <Footer /> component", () =>{
    test('renders Footer component' , () => {
        render (<Footer />);
    })
    
    test('should renders 2 anchor tags' , () => {
        render (<Footer />);
        const Links = screen.getAllByRole('link');
        expect(Links.length).toBe(2);
    })
    
    test('should have URL about.html' , () => {
        render (<Footer />);
        const aboutLink = screen.getByRole('link' , { name: 'About Us'});
        expect(aboutLink).toHaveAttribute('href','aboutus.html');
        
    })
    
    test('should have URL Contact.html' , () => {
        render (<Footer />);
        const aboutLink = screen.getByRole('link' , { name: 'Contact Us'});
        expect(aboutLink).toHaveAttribute('href','contact.html');
        
    })
    
    test('should have classname' , () => {
        render (<Footer />);
        const footer = screen.getByRole('contentinfo');
        expect(footer).toHaveClass('footerWrapper');
        
    })
})
