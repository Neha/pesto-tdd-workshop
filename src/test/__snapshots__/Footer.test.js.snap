// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <Footer /> component Should match snapshot 1`] = `
<footer
  className="footerWrapper"
>
  <a
    href="aboutus.html"
  >
    About Us
  </a>
  <a
    href="contact.html"
  >
    Contact Us
  </a>
</footer>
`;
